COM-2597 Add Quotes Grid and create quote section on seller and buyer page

rajeshdatir opened the pull request.

ghost commented:
@rajesh-datir Since the quotes filter story is not done and may take a while to finish I need this PR rebased so:
* The COM 2595 commit is removed (it's already in the latest `feature/B2B-phase-2` commit).
* All COM 2598 (quote filters) commits removed from this PR. We can later rebase that story on this PR or `feature/B2B-phase-2` once this is merged in.
* The COM 2619 commits need to come before the commits from this story because otherwise I will have trouble resolving the merge conflicts.

michaelsadlonkibo commented on scripts/modules/b2b-account/quotes.js:
```diff
@@ -57,9 +91,89 @@ define([
+            },
+            {
+                index: 'submittedDate',
+                displayName: 'Created',
```
Should `submittedDate` really map to `Created` exactly? Seems it would be better to call this column `Submitted`. Can you double check the requirements here? Thanks.

rajeshdatir replied:
changed label to submittedDate

michaelsadlonkibo commented on scripts/modules/mozu-quotes-grid/mozuquotesgrid-view.js:
```diff
@@ -0,0 +1,95 @@
+                        if (this.model.get('autoload')) {
+                            self.model.setIndex(0);
+                        }
+                    } catch (error) {
```
Why an empty try/catch block? This is bad practice and should be removed.

rajeshdatir replied:
removed try-catch

michaelsadlonkibo commented on labels/de-DE.json:
```diff
@@ -86,6 +86,7 @@
   "creditAmount": "Kreditbetrag",
   "creditCard": "Kreditkarte",
   "creditDate": "Kredit Datum",
+  "createQuote": "Angebot erstellen",
```
This is in the wrong spot. Please place this in the right spot alphabetically. It should be after `createAccount` I think if capitals sort before lowercase. Otherwise after createdDate. You can check this online with an alphasort.

rajeshdatir replied:
Added at correct location

michaelsadlonkibo commented on labels/en-US.json:
```diff
@@ -127,6 +127,7 @@
   "createAccount": "Create an account for easy checkout on future orders",
   "createdBy": "Created By",
   "createdDate": "Created Date",
+  "createQuote": "Create Quote",
```
Same as above. Please place this in the right spot alphabetically. Use an alphasort to check.

rajeshdatir replied:
Added at correct location

michaelsadlonkibo commented on labels/en-US.json:
```diff
@@ -530,6 +531,7 @@
   "quoteHistory": "Quote History",
   "quoteName": "Quote Name",
   "quotePrint": "Print Quote",
+  "quotes": "Quotes",
```
This should go after quoteSummary not before. Same place in both files. Use an alphasort if unsure.

rajeshdatir replied:
Added at the correct location

rajeshdatir commented on scripts/modules/b2b-account/b2b-account.js:
```diff
@@ -122,7 +122,7 @@ define([
       // This call is gluttonous and should be replaced with a call to users with filter params for
       // the user IDs we need.
-      var b2bAccount = new B2BAccountModels.b2bAccount({id: require.mozuData('user').accountId});
+      var b2bAccount = new B2BAccountModels.b2bAccount({id: require.mozuData('user').accountId});     
```
Removed whitespaces

ghost commented:
@michaelsadlonkibo For Now I keep all commits as it is. Once you merge COM-2619 and other branches I will do squash merge for this branch. Please let me know once done

ghost commented:
I have rebased, cleaned up, and force pushed this PR since it wasn't done yet. We will need to go over and make sure nothing broke since there were a number of merge conflicts.

kibo-rajeshdatir committed:
COM-2597 B2B P2: Add Quotes Grid and Create quote section (UI) on seller and myaccount page

Added Quotes Grid on seller and buyer page
Integrated Quote grid
Added account search dropdown on seller page
Added sorting functionality on columns
Made some cosmetic changes

michaelsadlonkibo commented on scripts/modules/models-customer.js:
```diff
@@ -387,10 +389,15 @@
-            _.defer(function (cust) {
-                cust.getCards();
-            }, self);
+            if (window.location.pathname.toLowerCase().indexOf("selleraccount") < 0) {
```
Don't like this fix. It's a hack and I think we need to find a better solution.

You might try checking if the user is a Seller like how `isNonPurchaser` checks the behaviors. Maybe a `isSellerAccount` that checks `isSalesRep`.

Alternatively I would keep `EditableCustomer` for Buyers only and create a `EditableSeller` or `EditableSellerAccount` for use with Sellers only. Long-term that might be the better approach. If you can separate them without tons of work I think I would prefer that.

rajeshdatir replied:
added **if (!isSalesRep)** check

michaelsadlonkibo commented on stylesheets/modules/mozugrid/grid.less:
```diff
@@ -1,55 +1,71 @@
     width: 100%;
     margin: 15px 0;
     border: 1px solid #999;
+
```
So many whitespace changes. Please revert this is a public repo. Thanks.

rajeshdatir replied:
It is very difficult to revert  in vs2019. so reverted in notepad.
When  I added Bhushan's changes, It autoformats document on paste.  need to find solution to this problem

michaelsadlonkibo replied:
Thanks @rajeshdatir. I did this again since Bhushan's commit caused the same issue. @SharmaBhushan please be careful with white space/formatting only changes. I recommend you disable format on paste/save since it can reformat the entire file.

SharmaBhushan replied:
Sure @michaelsadlonkibo I will take care. Thanks

michaelsadlonkibo reviewed (changes requested):
Mostly looks good. Requested some changes. I'm still not a fan of creating a new grid component but don't know this code well enough to make another suggestion. It works.

Keep in mind we may need to share this new grid component with other pages so it needs to be reusable. Can discuss later.

kibo-rajeshdatir committed:
COM-2597 Make single call to b2baccount for common acc ids

Increased header name font size

SharmaBhushan committed:
COM-2597, COM-2930 Demo feedback Changes

Fix sorting, resize sort icon

Commit:
Revert whitespace/formatting changes
